server/routes/pokemonRoutes.js

```diff
@@ -58,24 +58,43 @@ module.exports = {
 
 							let formattedPokemon = [];
 
+							let evolveMap = {};
+
 							for(let i = 0; i < rawPokemon.length; i++){
 								let pokemon = rawPokemon[i];
 
+								let id = pokemon.pokemon_id.toString();
+
 								if(pokemon.hasOwnProperty('is_egg') && !pokemon.is_egg){
 									let candy = 0;
 									for(let j = 0; j < rawCandies.length; j++){
 										let rawCandy = rawCandies[j];
 
-										if(rawCandy.family_id.toString() === props.pokemonFamilyIdByPokedexNum[pokemon.pokemon_id.toString()]){
+										if(rawCandy.family_id.toString() === props.pokemonFamilyIdByPokedexNum[id]){
 											candy = rawCandy.candy;
 										}
 									}
 
-									let name = props.pokemonNamesByDexNum[pokemon.pokemon_id.toString()];
+									let name = props.pokemonNamesByDexNum[id];
 									if(pokemon.hasOwnProperty('nickname') && pokemon.nickname.length > 0){
 										name = pokemon.nickname;
 									}
 
+									let evolve = {'sort': 0, 'descendants': []};
+									if (id in evolveMap){
+										evolve = evolveMap[id]
+									} else {
+										props.pokemonEvolutionByDexNum[id].forEach(function(descendant){
+											let canEvolve = Math.trunc(candy / descendant.cost);
```

Comment:
This formula isn't correct. For each evolve you get 1 extra candy, so only the cost of the first evolve is the full descendant.cost the rest are 1 less, e.g. first Pidgey evolve costs 12 candy, the rest effectively cost 11 because of chaining that extra evolve candy.

tl;dr: the correct formula = (candy - 1) / (descendant.cost - 1)